### DISCUSSION TITLE
Software-triggering USB camera

### DISCUSSION THREAD

**Discussion by KostasTsar** [bonsai-rx/bonsai#1660] [Q&A]
Hello, apologies if this has been asked before, I was unable to find quite a similar question.

Basically I have a series of USB cameras and I would like to trigger a frame capture via a digital input from Bonsai. I can't figure out though how to pass a trigger to the Camera Capture node. Is something like this possible? I am curious how can this be done in general, but specificially the cameras I have in mind are : ELP 2MP AR0234 SENSOR 1200P/ 1080P 90FPS GLOBAL SHUTTER USB CAMERA.

If this logic is not possible, could you suggest an alternative?

Thank you in advance.

**Comment by bruno-f-cruz** [comment]
Heya!

Not sure I completely understood your question. If you want to trigger a frame via a digital input, you need a camera capable of being externally triggered and something that Bonsai can control to generate that trigger (e.g. an Arduino). The camera you mention doesn't seem to have any GPIO capabilities so I am not sure sure if this is an option. If you just want to get the latest camera frame after an event from Bonsai, you can use `Sample`, `CombineLatest` or `WithLatestFrom` combinators.

I can expand a bit further but I am not sure what you are trying to achieve exactly!

**Reply by KostasTsar** [reply]
Hi! Thank you for the answer, makes sense. I guess the question really was, can I somehow synchronously trigger cameras that do not have GPIO from Bonsai. Acquiring the frames with a combinator would not be quite the same.
Will look into alternative cameras most likely, thank you.
